File: app/src/main/java/top/mikoto/sangnam/Adapters/AlarmListViewAdapter.java
package top.mikoto.sangnam.Adapters;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.text.Html;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import top.mikoto.sangnam.Activities.AddAlarmActivity;
import top.mikoto.sangnam.Activities.MainActivity;
import top.mikoto.sangnam.Models.AlarmModel;
import top.mikoto.sangnam.R;
import top.mikoto.sangnam.Utils.AlarmManager;

import static top.mikoto.sangnam.Utils.AlarmManager.parseDays;
import static top.mikoto.sangnam.Utils.AlarmManager.parseTime;

public class AlarmListViewAdapter extends ArrayAdapter<AlarmModel> {
    private ArrayList<AlarmModel> itemArrayList = new ArrayList<>();
    private Context context;
    private int rsrc;

    public AlarmListViewAdapter(@NonNull Context context, int resource, @NonNull List<AlarmModel> objects) {
        super(context, resource, objects);
        itemArrayList = (ArrayList<AlarmModel>) objects;
        this.context = context;
        rsrc = resource;
    }

    @Override
    public int getCount() {
        Log.d("MisakaMOE","size : "+itemArrayList.size());
        return itemArrayList.size();
    }

    @Override
    public AlarmModel getItem(int i) {

        return itemArrayList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View v = view;
        if(v == null)
        {
            LayoutInflater li = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            v = li.inflate(rsrc, null);
        }
        Log.d("MisakaMOE","getView called");
        Objects.requireNonNull(v).setLongClickable(true);

        TextView txtTime = v.findViewById(R.id.txtTime);
        TextView txtDay = v.findViewById(R.id.txtDay);
        Switch sw = v.findViewById(R.id.sw);

        AlarmModel item = itemArrayList.get(i);
        if(item != null)
        {
            AlarmManager alarmManager = new AlarmManager(context);

            txtTime.setText(parseTime(item.getTime()));
            txtDay.setText(Html.fromHtml(parseDays(item.getDays())));
            sw.setChecked((item.getRun()!=0)); //if run was 0, set false.

            sw.setOnCheckedChangeListener((buttonView, isChecked) -> {
                if(isChecked)
                {
                    item.setRun(1);
                }
                else
                {
                    item.setRun(0);
                }
                alarmManager.updateAlarm(item);
            });

            v.setOnClickListener(vi -> {
                Intent intent = new Intent(vi.getContext(), AddAlarmActivity.class);
                intent.putExtra("_id", item.get_id());
                ((MainActivity) context).startActivityForResult(intent,200);
            });

        }

        return v;
    }

    public void addItem(String time, String day) {
        AlarmModel item = new AlarmModel();
        item.setTime(time);
        item.setDays(day);
        item.setRun(1);

        itemArrayList.add(item);
    }



}

File: app/src/main/java/top/mikoto/sangnam/adapters/AlarmListViewAdapter.kt
package top.mikoto.sangnam.adapters

import android.content.Context
import android.content.Intent
import android.text.Html
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.CompoundButton
import android.widget.Switch
import android.widget.TextView
import top.mikoto.sangnam.R
import top.mikoto.sangnam.activities.AddAlarmActivity
import top.mikoto.sangnam.activities.MainActivity
import top.mikoto.sangnam.models.AlarmModel
import top.mikoto.sangnam.utils.AlarmManager
import java.util.*

class AlarmListViewAdapter(context: Context, resource: Int, objects: List<AlarmModel>) : ArrayAdapter<AlarmModel>(context, resource, objects) {
    private val itemArrayList: ArrayList<AlarmModel> = objects as ArrayList<AlarmModel>
    override fun getCount(): Int {
        Log.d("MisakaMOE", "size : " + itemArrayList.size)
        return itemArrayList.size
    }

    override fun getItem(i: Int): AlarmModel {
        return itemArrayList[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }

    override fun getView(i: Int, view: View, viewGroup: ViewGroup): View {
        Log.d("MisakaMOE", "getView called")
        Objects.requireNonNull(view).isLongClickable = true
        val txtTime = view.findViewById<TextView>(R.id.txtTime)
        val txtDay = view.findViewById<TextView>(R.id.txtDay)
        val sw = view.findViewById<Switch>(R.id.sw)
        val item = itemArrayList[i]
        val alarmManager = AlarmManager(context)
        txtTime.text = AlarmManager.parseTime(item.time)
        txtDay.text = Html.fromHtml(AlarmManager.parseDays(item.days))
        sw.isChecked = item.run != 0 //if run was 0, set false.
        sw.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) {
                item.run = 1
            } else {
                item.run = 0
            }
            alarmManager.updateAlarm(item)
        }
        view.setOnClickListener { vi: View ->
            val intent = Intent(vi.context, AddAlarmActivity::class.java)
            intent.putExtra("_id", item._id)
            (context as MainActivity).startActivityForResult(intent, 200)
        }
        return view
    }

    fun addItem(time: String?, day: String?) {
        val item = AlarmModel()
        item.time = time
        item.days = day
        item.run = 1
        itemArrayList.add(item)
    }

}